src/core/server/path.ts

```diff
@@ -18,34 +18,53 @@
  */
 
 import { join } from 'path';
-import { accessSync, R_OK } from 'fs';
-import { find } from 'lodash';
-import { fromRoot } from '../../utils';
+import { accessSync, constants } from 'fs';
+import { TypeOf, schema } from '@kbn/config-schema';
+import { fromRoot } from '../utils';
+
+const isString = (v: any): v is string => typeof v === 'string';
 
 const CONFIG_PATHS = [
   process.env.KIBANA_PATH_CONF && join(process.env.KIBANA_PATH_CONF, 'kibana.yml'),
-  process.env.CONFIG_PATH, //deprecated
+  process.env.CONFIG_PATH, // deprecated
   fromRoot('config/kibana.yml'),
-  '/etc/kibana/kibana.yml'
-].filter(Boolean);
+  '/etc/kibana/kibana.yml',
+].filter(isString);
 
 const DATA_PATHS = [
-  process.env.DATA_PATH, //deprecated
+  process.env.DATA_PATH, // deprecated
   fromRoot('data'),
-  '/var/lib/kibana'
-].filter(Boolean);
+  '/var/lib/kibana',
+].filter(isString);
 
-function findFile(paths) {
-  const availablePath = find(paths, configPath => {
+function findFile(paths: string[]) {
+  const availablePath = paths.find(configPath => {
     try {
-      accessSync(configPath, R_OK);
+      accessSync(configPath, constants.R_OK);
       return true;
     } catch (e) {
-      //Check the next path
+      // Check the next path
     }
   });
   return availablePath || paths[0];
 }
 
-export const getConfig = () => findFile(CONFIG_PATHS);
-export const getData = () => findFile(DATA_PATHS);
+/**
+ * Get the path where the config files are stored
+ * @internal
+ */
+export const getConfigPath = () => findFile(CONFIG_PATHS);
+/**
+ * Get the path where the data can be stored
+ * @internal
+ */
+export const getDataPath = () => findFile(DATA_PATHS);
+
+export type PathConfigType = TypeOf<typeof config.schema>;
+
+export const config = {
+  path: 'path',
+  schema: schema.object({
+    data: schema.string({ defaultValue: () => getDataPath() }),
```

Comment:
Since `config` is used in some plugins as well (telemetry), should we expose it?

Reply:
> Since config is used in some plugins as well (telemetry), should we expose it?

According to the comment, they need only value. So yes, let's expose it via contract.
https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/telemetry/index.ts#L49